src/OpenSEE/Scripts/TSX/store/dataSlice.tsx

```diff
@@ -945,14 +945,18 @@ function getData(key: OpenSee.IGraphProps, dispatch: any, options: OpenSee.IAnal
                 async: true
             });
 
+            handlePOW.then((data) => {
+                dispatch(AddData({ key: key, data: data.Data, requestID: requestID, secondary: false }));
+                dispatch(InitiateDetailed({ DataType: key.DataType, EventId: key.EventId, NoCompress: key.NoCompress, DataGrouping: "Time" }));
+            })
+
             handleFreq.then((data) => {
                 dispatch(AddData({ key: key, data: data.Data, requestID: requestID, secondary: true }));
-                dispatch(InitiateDetailed(key));
+                dispatch(InitiateDetailed({ DataType: key.DataType, EventId: key.EventId, NoCompress: key.NoCompress, DataGrouping: "Freq" }));
```

Comment:
is there a reason we added the `DataGrouping` to `IGraphProps`? My recollection is `Graph` actually represents a single Plot (e.g. the Voltage Plot) so it has a `Time` and a `Freq` component. The `IGraphProps` interface should correspond to the `Props` for it.
Looks like you made `DataGrouping` optional so it doesn't throw exceptions when building, but It becomes confusing since that type no longer represents the `Props` for a `Graph` object

Reply:
It's just that I needed to pass it through to the `getDetailedData()` function. I had considered that it's probably more correct to introduce another interface or use an inline data type, but I figured I'd submit it this way and let you comment.

Reply:
I got rid of `DataGrouping` altogether by waiting on all non-detailed data to come in before triggering the queries for detailed data. Then the `getDetailedData()` functoin just needs to trigger both Time and Freq queries simultaneously like the non-detailed one does.